utils/gridlink_impl.c.src

```diff
@@ -369,7 +362,10 @@ cellarray_DOUBLE * gridlink_DOUBLE(const int64_t NPART,
                 SGLIB_ARRAY_ELEMENTS_EXCHANGER(DOUBLE, first->x, i, j); \
                 SGLIB_ARRAY_ELEMENTS_EXCHANGER(DOUBLE, first->y, i, j); \
                 SGLIB_ARRAY_ELEMENTS_EXCHANGER(DOUBLE, first->z, i, j); \
-                SGLIB_ARRAY_ELEMENTS_EXCHANGER(int64_t, first->original_index, i, j); \
+                /* original_index is not used if copying the particles */ \
+                if(!options->copy_particles) { \
```

Comment:
And another candidate - will you please change to `options->copy_particles == 0`